src/plugins/kibana_utils/public/state_management/url/kbn_url_storage.ts

```diff
@@ -15,30 +15,39 @@ import { replaceUrlHashQuery, replaceUrlQuery } from './format';
 import { url as urlUtils } from '../../../common';
 
 /**
- * Parses a kibana url and retrieves all the states encoded into url,
+ * Parses a kibana url and retrieves all the states encoded into the URL,
  * Handles both expanded rison state and hashed state (where the actual state stored in sessionStorage)
  * e.g.:
  *
  * given an url:
  * http://localhost:5601/oxf/app/kibana#/yourApp?_a=(tab:indexedFields)&_b=(f:test,i:'',l:'')
  * will return object:
  * {_a: {tab: 'indexedFields'}, _b: {f: 'test', i: '', l: ''}};
+ *
+ *
+ * By default due to Kibana legacy reasons assumed that state is stored in a query inside a hash part of the URL:
+ * http://localhost:5601/oxf/app/kibana#/yourApp?_a={STATE}
+ *
+ * { getFromHashQuery: false } option should be used in case state is stored in a main query (not in a hash):
+ * http://localhost:5601/oxf/app/kibana?_a={STATE}#/yourApp
+ *
  */
-export function getStatesFromKbnUrl(
+export function getStatesFromKbnUrl<State extends object = Record<string, unknown>>(
   url: string = window.location.href,
-  keys?: string[]
-): Record<string, unknown> {
-  const query = parseUrlHash(url)?.query;
+  keys?: Array<keyof State>,
+  { getFromHashQuery = true }: { getFromHashQuery: boolean } = { getFromHashQuery: true }
```

Comment:
Needed to extend the functionality of this helper because the example plugin doesn't use hash routing and stores state not inside the hash part of the URL but inside of the `location.search` instead


`setStateToKbnURL` already had this param in place